// angular-toastify/bundles/angular-toastify.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('angular-toastify', ['exports', '@angular/core', '@angular/common'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['angular-toastify'] = {}, global.ng.core, global.ng.common));
}(this, (function (exports, i0, common) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    var toastIdentitySequence = 0;
    var Toast = /** @class */ (function () {
        function Toast(message, type) {
            this.message = message;
            this.type = type;
            this.time = new Date().getTime();
            this.id = toastIdentitySequence++;
        }
        return Toast;
    }());

    var ToastType;
    (function (ToastType) {
        ToastType[ToastType["info"] = 0] = "info";
        ToastType[ToastType["success"] = 1] = "success";
        ToastType[ToastType["warning"] = 2] = "warning";
        ToastType[ToastType["error"] = 3] = "error";
        ToastType[ToastType["default"] = 4] = "default";
    })(ToastType || (ToastType = {}));

    var ToastService = /** @class */ (function () {
        function ToastService() {
            this.toastAddedEvent = new i0.EventEmitter();
            this.dismissAllEvent = new i0.EventEmitter();
        }
        ToastService.prototype.dismissAll = function () {
            this.dismissAllEvent.emit();
        };
        ToastService.prototype.info = function (message) {
            var toast = new Toast(message, ToastType.info);
            this.toastAddedEvent.emit(toast);
        };
        ToastService.prototype.success = function (message) {
            var toast = new Toast(message, ToastType.success);
            this.toastAddedEvent.emit(toast);
        };
        ToastService.prototype.warn = function (message) {
            var toast = new Toast(message, ToastType.warning);
            this.toastAddedEvent.emit(toast);
        };
        ToastService.prototype.error = function (message) {
            var toast = new Toast(message, ToastType.error);
            this.toastAddedEvent.emit(toast);
        };
        ToastService.prototype.default = function (message) {
            var toast = new Toast(message, ToastType.default);
            this.toastAddedEvent.emit(toast);
        };
        return ToastService;
    }());
    /** @nocollapse */ ToastService.ɵprov = i0__namespace.ɵɵdefineInjectable({ factory: function ToastService_Factory() { return new ToastService(); }, token: ToastService, providedIn: "root" });
    ToastService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    ToastService.ctorParameters = function () { return []; };

    var ToastifyToastContainerComponent = /** @class */ (function () {
        function ToastifyToastContainerComponent(_toastService, _cd) {
            this._toastService = _toastService;
            this._cd = _cd;
            this.transitionDurations = 400;
            this.position = 'top-right';
            this.transition = 'bounce';
            this.autoClose = 5000;
            this.hideProgressBar = false;
            this.pauseOnHover = true;
            this.pauseOnVisibilityChange = true;
            this.closeOnClick = true;
            this.newestOnTop = false;
            this.iconLibrary = 'none';
            this.ToastType = ToastType;
            this.toasts = new Array();
            this.toastTransitionDict = {};
        }
        ToastifyToastContainerComponent.prototype.ngOnChanges = function () {
            this._cd.markForCheck();
        };
        ToastifyToastContainerComponent.prototype.dismiss = function (toast) {
            var _this = this;
            this.toastTransitionDict[toast.id] = TransitionState.exiting;
            setTimeout(function () {
                var index = _this.toasts.indexOf(toast);
                _this.toasts.splice(index, 1);
                _this._cd.markForCheck();
            }, this.transitionDurations);
        };
        ToastifyToastContainerComponent.prototype.getClass = function (toast) {
            var base = "toast toast--" + ToastType[toast.type] + " ";
            var state = this.toastTransitionDict[toast.id];
            if (state === TransitionState.entering) {
                base += this.transition + "-enter " + this.transition + "-enter--" + this.position;
            }
            else if (state === TransitionState.exiting) {
                base += this.transition + "-exit " + this.transition + "-exit--" + this.position;
            }
            return base;
        };
        ToastifyToastContainerComponent.prototype.ngOnInit = function () {
            var _this = this;
            this._toastService.dismissAllEvent.subscribe(function () {
                _this.toasts = new Array();
                _this._cd.markForCheck();
            });
            this._toastService.toastAddedEvent.subscribe(function (toast) {
                _this.toastTransitionDict[toast.id] = TransitionState.entering;
                setTimeout(function () {
                    _this.toastTransitionDict[toast.id] = TransitionState.noTransition;
                    _this._cd.markForCheck();
                }, _this.transitionDurations);
                if (_this.newestOnTop) {
                    _this.toasts.unshift(toast);
                }
                else {
                    _this.toasts.push(toast);
                }
                _this._cd.markForCheck();
            });
        };
        return ToastifyToastContainerComponent;
    }());
    ToastifyToastContainerComponent.decorators = [
        { type: i0.Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'lib-toastify-toast-container',
                    template: "<div class=\"toast-container toast-container--{{position}}\">\n    <lib-toastify-toast *ngFor=\"let toast of toasts\" [class]=\"getClass(toast)\" style=\"animation-fill-mode: both; animation-duration: 0.75s\"\n        [autoClose]=\"autoClose\" [toast]=\"toast\" (dismissEvent)=\"dismiss(toast)\" [hideProgressBar]=\"hideProgressBar\" [pauseOnHover]=\"pauseOnHover\"\n        [pauseOnVisibilityChange]=\"pauseOnVisibilityChange\" [closeOnClick]=\"closeOnClick\" [iconLibrary]=\"iconLibrary\">\n    </lib-toastify-toast>\n</div>",
                    changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    styles: [".toast-container{z-index:9999;-webkit-transform:translateZ(9999px);position:fixed;padding:4px;width:320px;box-sizing:border-box;color:#fff}.toast-container--top-left{top:1em;left:1em}.toast-container--top-center{top:1em;left:50%;margin-left:-160px}.toast-container--top-right{top:1em;right:1em}.toast-container--bottom-left{bottom:1em;left:1em}.toast-container--bottom-center{bottom:1em;left:50%;margin-left:-160px}.toast-container--bottom-right{bottom:1em;right:1em}@media only screen and (max-width:480px){.toast-container{width:100vw;padding:0;left:0;margin:0}.toast-container--top-center,.toast-container--top-left,.toast-container--top-right{top:0}.toast-container--bottom-center,.toast-container--bottom-left,.toast-container--bottom-right{bottom:0}.toast-container--rtl{right:0;left:auto}}.toast{position:relative;min-height:64px;box-sizing:border-box;margin-bottom:1rem;padding:8px;border-radius:1px;box-shadow:0 1px 10px 0 rgba(0,0,0,.1),0 2px 15px 0 rgba(0,0,0,.05);display:flex;justify-content:space-between;max-height:800px;overflow:hidden;font-family:sans-serif;cursor:pointer;direction:ltr}.toast--rtl{direction:rtl}.toast--default{background:#fff;color:#aaa}.toast--info{background:#3498db}.toast--success{background:#07bc0c}.toast--warning{background:#f1c40f}.toast--error{background:#e74c3c}.toast-body{margin:auto 0;flex:1}@media only screen and (max-width:480px){.toast{margin-bottom:0}}@-webkit-keyframes bounceInRight{0%,60%,75%,90%,to{-webkit-animation-timing-function:cubic-bezier(.215,.61,.355,1);animation-timing-function:cubic-bezier(.215,.61,.355,1)}0%{opacity:0;transform:translate3d(3000px,0,0)}60%{opacity:1;transform:translate3d(-25px,0,0)}75%{transform:translate3d(10px,0,0)}90%{transform:translate3d(-5px,0,0)}to{transform:none}}@keyframes bounceInRight{0%,60%,75%,90%,to{-webkit-animation-timing-function:cubic-bezier(.215,.61,.355,1);animation-timing-function:cubic-bezier(.215,.61,.355,1)}0%{opacity:0;transform:translate3d(3000px,0,0)}60%{opacity:1;transform:translate3d(-25px,0,0)}75%{transform:translate3d(10px,0,0)}90%{transform:translate3d(-5px,0,0)}to{transform:none}}@-webkit-keyframes bounceOutRight{20%{opacity:1;transform:translate3d(-20px,0,0)}to{opacity:0;transform:translate3d(2000px,0,0)}}@keyframes bounceOutRight{20%{opacity:1;transform:translate3d(-20px,0,0)}to{opacity:0;transform:translate3d(2000px,0,0)}}@-webkit-keyframes bounceInLeft{0%,60%,75%,90%,to{-webkit-animation-timing-function:cubic-bezier(.215,.61,.355,1);animation-timing-function:cubic-bezier(.215,.61,.355,1)}0%{opacity:0;transform:translate3d(-3000px,0,0)}60%{opacity:1;transform:translate3d(25px,0,0)}75%{transform:translate3d(-10px,0,0)}90%{transform:translate3d(5px,0,0)}to{transform:none}}@keyframes bounceInLeft{0%,60%,75%,90%,to{-webkit-animation-timing-function:cubic-bezier(.215,.61,.355,1);animation-timing-function:cubic-bezier(.215,.61,.355,1)}0%{opacity:0;transform:translate3d(-3000px,0,0)}60%{opacity:1;transform:translate3d(25px,0,0)}75%{transform:translate3d(-10px,0,0)}90%{transform:translate3d(5px,0,0)}to{transform:none}}@-webkit-keyframes bounceOutLeft{20%{opacity:1;transform:translate3d(20px,0,0)}to{opacity:0;transform:translate3d(-2000px,0,0)}}@keyframes bounceOutLeft{20%{opacity:1;transform:translate3d(20px,0,0)}to{opacity:0;transform:translate3d(-2000px,0,0)}}@-webkit-keyframes bounceInUp{0%,60%,75%,90%,to{-webkit-animation-timing-function:cubic-bezier(.215,.61,.355,1);animation-timing-function:cubic-bezier(.215,.61,.355,1)}0%{opacity:0;transform:translate3d(0,3000px,0)}60%{opacity:1;transform:translate3d(0,-20px,0)}75%{transform:translate3d(0,10px,0)}90%{transform:translate3d(0,-5px,0)}to{transform:translateZ(0)}}@keyframes bounceInUp{0%,60%,75%,90%,to{-webkit-animation-timing-function:cubic-bezier(.215,.61,.355,1);animation-timing-function:cubic-bezier(.215,.61,.355,1)}0%{opacity:0;transform:translate3d(0,3000px,0)}60%{opacity:1;transform:translate3d(0,-20px,0)}75%{transform:translate3d(0,10px,0)}90%{transform:translate3d(0,-5px,0)}to{transform:translateZ(0)}}@-webkit-keyframes bounceOutUp{20%{transform:translate3d(0,-10px,0)}40%,45%{opacity:1;transform:translate3d(0,20px,0)}to{opacity:0;transform:translate3d(0,-2000px,0)}}@keyframes bounceOutUp{20%{transform:translate3d(0,-10px,0)}40%,45%{opacity:1;transform:translate3d(0,20px,0)}to{opacity:0;transform:translate3d(0,-2000px,0)}}@-webkit-keyframes bounceInDown{0%,60%,75%,90%,to{-webkit-animation-timing-function:cubic-bezier(.215,.61,.355,1);animation-timing-function:cubic-bezier(.215,.61,.355,1)}0%{opacity:0;transform:translate3d(0,-3000px,0)}60%{opacity:1;transform:translate3d(0,25px,0)}75%{transform:translate3d(0,-10px,0)}90%{transform:translate3d(0,5px,0)}to{transform:none}}@keyframes bounceInDown{0%,60%,75%,90%,to{-webkit-animation-timing-function:cubic-bezier(.215,.61,.355,1);animation-timing-function:cubic-bezier(.215,.61,.355,1)}0%{opacity:0;transform:translate3d(0,-3000px,0)}60%{opacity:1;transform:translate3d(0,25px,0)}75%{transform:translate3d(0,-10px,0)}90%{transform:translate3d(0,5px,0)}to{transform:none}}@-webkit-keyframes bounceOutDown{20%{transform:translate3d(0,10px,0)}40%,45%{opacity:1;transform:translate3d(0,-20px,0)}to{opacity:0;transform:translate3d(0,2000px,0)}}@keyframes bounceOutDown{20%{transform:translate3d(0,10px,0)}40%,45%{opacity:1;transform:translate3d(0,-20px,0)}to{opacity:0;transform:translate3d(0,2000px,0)}}.bounce-enter--bottom-left,.bounce-enter--top-left{-webkit-animation-name:bounceInLeft;animation-name:bounceInLeft}.bounce-enter--bottom-right,.bounce-enter--top-right{-webkit-animation-name:bounceInRight;animation-name:bounceInRight}.bounce-enter--top-center{-webkit-animation-name:bounceInDown;animation-name:bounceInDown}.bounce-enter--bottom-center{-webkit-animation-name:bounceInUp;animation-name:bounceInUp}.bounce-exit--bottom-left,.bounce-exit--top-left{-webkit-animation-name:bounceOutLeft;animation-name:bounceOutLeft}.bounce-exit--bottom-right,.bounce-exit--top-right{-webkit-animation-name:bounceOutRight;animation-name:bounceOutRight}.bounce-exit--top-center{-webkit-animation-name:bounceOutUp;animation-name:bounceOutUp}.bounce-exit--bottom-center{-webkit-animation-name:bounceOutDown;animation-name:bounceOutDown}@-webkit-keyframes zoomIn{0%{opacity:0;transform:scale3d(.3,.3,.3)}50%{opacity:1}}@keyframes zoomIn{0%{opacity:0;transform:scale3d(.3,.3,.3)}50%{opacity:1}}@-webkit-keyframes zoomOut{0%{opacity:1}50%{opacity:0;transform:scale3d(.3,.3,.3)}to{opacity:0}}@keyframes zoomOut{0%{opacity:1}50%{opacity:0;transform:scale3d(.3,.3,.3)}to{opacity:0}}.zoom-enter{-webkit-animation-name:zoomIn;animation-name:zoomIn}.zoom-exit{-webkit-animation-name:zoomOut;animation-name:zoomOut}@-webkit-keyframes flipIn{0%{transform:perspective(400px) rotateX(90deg);-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;opacity:0}40%{transform:perspective(400px) rotateX(-20deg);-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in}60%{transform:perspective(400px) rotateX(10deg);opacity:1}80%{transform:perspective(400px) rotateX(-5deg)}to{transform:perspective(400px)}}@keyframes flipIn{0%{transform:perspective(400px) rotateX(90deg);-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;opacity:0}40%{transform:perspective(400px) rotateX(-20deg);-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in}60%{transform:perspective(400px) rotateX(10deg);opacity:1}80%{transform:perspective(400px) rotateX(-5deg)}to{transform:perspective(400px)}}@-webkit-keyframes flipOut{0%{transform:perspective(400px)}30%{transform:perspective(400px) rotateX(-20deg);opacity:1}to{transform:perspective(400px) rotateX(90deg);opacity:0}}@keyframes flipOut{0%{transform:perspective(400px)}30%{transform:perspective(400px) rotateX(-20deg);opacity:1}to{transform:perspective(400px) rotateX(90deg);opacity:0}}.flip-enter{-webkit-animation-name:flipIn;animation-name:flipIn}.flip-exit{-webkit-animation-name:flipOut;animation-name:flipOut}@-webkit-keyframes slideInRight{0%{transform:translate3d(110%,0,0);visibility:visible}to{transform:translateZ(0)}}@keyframes slideInRight{0%{transform:translate3d(110%,0,0);visibility:visible}to{transform:translateZ(0)}}@-webkit-keyframes slideInLeft{0%{transform:translate3d(-110%,0,0);visibility:visible}to{transform:translateZ(0)}}@keyframes slideInLeft{0%{transform:translate3d(-110%,0,0);visibility:visible}to{transform:translateZ(0)}}@-webkit-keyframes slideInUp{0%{transform:translate3d(0,110%,0);visibility:visible}to{transform:translateZ(0)}}@keyframes slideInUp{0%{transform:translate3d(0,110%,0);visibility:visible}to{transform:translateZ(0)}}@-webkit-keyframes slideInDown{0%{transform:translate3d(0,-110%,0);visibility:visible}to{transform:translateZ(0)}}@keyframes slideInDown{0%{transform:translate3d(0,-110%,0);visibility:visible}to{transform:translateZ(0)}}@-webkit-keyframes slideOutRight{0%{transform:translateZ(0)}to{visibility:hidden;transform:translate3d(110%,0,0)}}@keyframes slideOutRight{0%{transform:translateZ(0)}to{visibility:hidden;transform:translate3d(110%,0,0)}}@-webkit-keyframes slideOutLeft{0%{transform:translateZ(0)}to{visibility:hidden;transform:translate3d(-110%,0,0)}}@keyframes slideOutLeft{0%{transform:translateZ(0)}to{visibility:hidden;transform:translate3d(-110%,0,0)}}@-webkit-keyframes slideOutDown{0%{transform:translateZ(0)}to{visibility:hidden;transform:translate3d(0,500px,0)}}@keyframes slideOutDown{0%{transform:translateZ(0)}to{visibility:hidden;transform:translate3d(0,500px,0)}}@-webkit-keyframes slideOutUp{0%{transform:translateZ(0)}to{visibility:hidden;transform:translate3d(0,-500px,0)}}@keyframes slideOutUp{0%{transform:translateZ(0)}to{visibility:hidden;transform:translate3d(0,-500px,0)}}.slide-enter--bottom-left,.slide-enter--top-left{-webkit-animation-name:slideInLeft;animation-name:slideInLeft}.slide-enter--bottom-right,.slide-enter--top-right{-webkit-animation-name:slideInRight;animation-name:slideInRight}.slide-enter--top-center{-webkit-animation-name:slideInDown;animation-name:slideInDown}.slide-enter--bottom-center{-webkit-animation-name:slideInUp;animation-name:slideInUp}.slide-exit--bottom-left,.slide-exit--top-left{-webkit-animation-name:slideOutLeft;animation-name:slideOutLeft}.slide-exit--bottom-right,.slide-exit--top-right{-webkit-animation-name:slideOutRight;animation-name:slideOutRight}.slide-exit--top-center{-webkit-animation-name:slideOutUp;animation-name:slideOutUp}.slide-exit--bottom-center{-webkit-animation-name:slideOutDown;animation-name:slideOutDown}"]
                },] }
    ];
    /** @nocollapse */
    ToastifyToastContainerComponent.ctorParameters = function () { return [
        { type: ToastService },
        { type: i0.ChangeDetectorRef }
    ]; };
    ToastifyToastContainerComponent.propDecorators = {
        position: [{ type: i0.Input }],
        transition: [{ type: i0.Input }],
        autoClose: [{ type: i0.Input }],
        hideProgressBar: [{ type: i0.Input }],
        pauseOnHover: [{ type: i0.Input }],
        pauseOnVisibilityChange: [{ type: i0.Input }],
        closeOnClick: [{ type: i0.Input }],
        newestOnTop: [{ type: i0.Input }],
        iconLibrary: [{ type: i0.Input }]
    };
    var TransitionState;
    (function (TransitionState) {
        TransitionState[TransitionState["entering"] = 0] = "entering";
        TransitionState[TransitionState["noTransition"] = 1] = "noTransition";
        TransitionState[TransitionState["exiting"] = 2] = "exiting";
    })(TransitionState || (TransitionState = {}));

    var ToastifyToastComponent = /** @class */ (function () {
        function ToastifyToastComponent(_cd, _zone) {
            this._cd = _cd;
            this._zone = _zone;
            this.autoClose = 5000;
            this.hideProgressBar = false;
            this.pauseOnHover = true;
            this.pauseOnVisibilityChange = true;
            this.closeOnClick = true;
            this.dismissEvent = new i0.EventEmitter();
            this.ToastType = ToastType;
            this.running = false;
        }
        ToastifyToastComponent.prototype.ngOnInit = function () {
            this.autoCloseRemaining = this.autoClose;
            this.startTime = this.toast.time;
            // Do not start timer when toast is prompted while window is out of focus
            if (this.handleVisibilityChange && document.visibilityState === 'visible') {
                this.startCloseTimer();
            }
        };
        ToastifyToastComponent.prototype.ngOnDestroy = function () {
            this.clearTimerTimeout();
        };
        ToastifyToastComponent.prototype.startCloseTimer = function () {
            var _this = this;
            if (this.running || !this.autoClose) {
                return;
            }
            this.startTime = new Date().getTime();
            this.running = true;
            this.autoDismissTimeout = this._zone.runOutsideAngular(function () { return setTimeout(function () {
                _this._zone.run(function () {
                    _this.dismissEvent.emit();
                    _this._cd.markForCheck();
                });
            }, _this.autoCloseRemaining); });
        };
        ToastifyToastComponent.prototype.pauseCloseTimer = function () {
            this.running = false;
            this.clearTimerTimeout();
            // Calculate the elapsed time, subtract remaining time
            this.pauseTime = new Date().getTime();
            var elapsed = this.pauseTime - this.startTime;
            this.autoCloseRemaining -= elapsed;
        };
        ToastifyToastComponent.prototype.clearTimerTimeout = function () {
            if (this.autoDismissTimeout !== undefined) {
                clearTimeout(this.autoDismissTimeout);
            }
        };
        ToastifyToastComponent.prototype.handleDismissButtonAction = function () {
            if (this.closeOnClick) {
                return; // Let the other event handle the dismissal
            }
            this.clearTimerTimeout();
            this.dismissEvent.emit();
        };
        ToastifyToastComponent.prototype.handleHostClick = function () {
            if (this.closeOnClick) {
                this.clearTimerTimeout();
                this.dismissEvent.emit();
            }
        };
        ToastifyToastComponent.prototype.handleMouseEnter = function () {
            if (this.pauseOnHover) {
                this.pauseCloseTimer();
            }
        };
        ToastifyToastComponent.prototype.handleMouseLeave = function () {
            if (this.pauseOnHover) {
                this.startCloseTimer();
            }
        };
        ToastifyToastComponent.prototype.handleVisibilityChange = function () {
            if (!this.pauseOnVisibilityChange) {
                return;
            }
            if (document.visibilityState !== 'visible') {
                this.pauseCloseTimer();
                this._cd.detectChanges();
            }
            else {
                this.startCloseTimer();
            }
        };
        return ToastifyToastComponent;
    }());
    ToastifyToastComponent.decorators = [
        { type: i0.Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'lib-toastify-toast',
                    template: "<div role=\"alert\" class=\"toast-body\">\n    <div class=\"icon-container\">\n        <span *ngIf=\"iconLibrary == 'material'\" [ngSwitch]=\"toast.type\">\n            <i class=\"material-icons\" *ngSwitchCase=\"ToastType.info\">info_outline</i>\n            <i class=\"material-icons\" *ngSwitchCase=\"ToastType.default\">info_outline</i>\n            <i class=\"material-icons\" *ngSwitchCase=\"ToastType.warning\">warning_outline</i>\n            <i class=\"material-icons\" *ngSwitchCase=\"ToastType.error\">error_outline</i>\n            <i class=\"material-icons\" *ngSwitchCase=\"ToastType.success\">done</i>\n        </span>\n        <span *ngIf=\"iconLibrary == 'font-awesome'\" [ngSwitch]=\"toast.type\">\n            <i *ngSwitchCase=\"ToastType.info\" class=\"fa fa-info\"></i>\n            <i *ngSwitchCase=\"ToastType.default\" class=\"fa fa-info\"></i>\n            <i *ngSwitchCase=\"ToastType.warning\" class=\"fa fa-exclamation-triangle\"></i>\n            <i *ngSwitchCase=\"ToastType.error\" class=\"fa fa-exclamation\"></i>\n            <i *ngSwitchCase=\"ToastType.success\" class=\"fa fa-check\"></i>\n        </span>\n    </div>\n    <div class=\"toast-container\">\n        <span>{{toast.message}}</span>\n    </div>\n</div>\n<button (click)=\"handleDismissButtonAction()\" class=\"close-button close-button--{{ToastType[toast.type]}}\" type=\"button\" aria-label=\"close\">\u2716</button>\n<div *ngIf=\"!hideProgressBar\" class=\"progress-bar progress-bar--{{ToastType[toast.type]}}\"\n    [style.animation-play-state]=\"running? 'running' : 'paused'\" [style.animation-duration]=\"this.autoClose + 'ms'\" style=\"opacity: 1;\"></div>",
                    changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    styles: [".toast{position:relative;min-height:64px;box-sizing:border-box;margin-bottom:1rem;padding:8px;border-radius:1px;box-shadow:0 1px 10px 0 rgba(0,0,0,.1),0 2px 15px 0 rgba(0,0,0,.05);display:flex;justify-content:space-between;max-height:800px;overflow:hidden;font-family:sans-serif;cursor:pointer;direction:ltr}.toast--rtl{direction:rtl}.toast--default{background:#fff;color:#aaa}.toast--info{background:#3498db}.toast--success{background:#07bc0c}.toast--warning{background:#f1c40f}.toast--error{background:#e74c3c}.toast-body{margin:auto 0;flex:1}@media only screen and (max-width:480px){.toast{margin-bottom:0}}.close-button{color:#fff;font-weight:700;font-size:14px;background:transparent;outline:none;border:none;padding:0;cursor:pointer;opacity:.7;transition:.3s ease;align-self:flex-start}.close-button--default{color:#000;opacity:.3}.close-button:focus,.close-button:hover{opacity:1}@-webkit-keyframes trackProgress{0%{transform:scaleX(1)}to{transform:scaleX(0)}}@keyframes trackProgress{0%{transform:scaleX(1)}to{transform:scaleX(0)}}.progress-bar{position:absolute;bottom:0;left:0;width:100%;height:5px;z-index:9999;opacity:.7;background-color:hsla(0,0%,100%,.7);transform-origin:left}.progress-bar,.progress-bar--animated{-webkit-animation:trackProgress linear 1 forwards;animation:trackProgress linear 1 forwards}.progress-bar--controlled{transition:transform .2s}.progress-bar--rtl{right:0;left:auto;transform-origin:right}.progress-bar--default{background:linear-gradient(90deg,#4cd964,#5ac8fa,#007aff,#34aadc,#5856d6,#ff2d55)}.icon-container,.toast-body{vertical-align:middle}.icon-container{display:inline-block;width:25px}.icon-container span{margin-right:.5rem;display:inline-block;text-align:center;width:20px}.icon-container span i{vertical-align:middle}.fa,.material-icons{font-size:18px}.toast-container{display:inline-block;width:calc(100% - 25px);vertical-align:middle}"]
                },] }
    ];
    /** @nocollapse */
    ToastifyToastComponent.ctorParameters = function () { return [
        { type: i0.ChangeDetectorRef },
        { type: i0.NgZone }
    ]; };
    ToastifyToastComponent.propDecorators = {
        autoClose: [{ type: i0.Input }],
        hideProgressBar: [{ type: i0.Input }],
        pauseOnHover: [{ type: i0.Input }],
        pauseOnVisibilityChange: [{ type: i0.Input }],
        closeOnClick: [{ type: i0.Input }],
        toast: [{ type: i0.Input }],
        iconLibrary: [{ type: i0.Input }],
        dismissEvent: [{ type: i0.Output }],
        handleHostClick: [{ type: i0.HostListener, args: ['click',] }],
        handleMouseEnter: [{ type: i0.HostListener, args: ['mouseenter',] }],
        handleMouseLeave: [{ type: i0.HostListener, args: ['mouseleave',] }],
        handleVisibilityChange: [{ type: i0.HostListener, args: ['document:visibilitychange',] }]
    };

    var AngularToastifyModule = /** @class */ (function () {
        function AngularToastifyModule() {
        }
        return AngularToastifyModule;
    }());
    AngularToastifyModule.decorators = [
        { type: i0.NgModule, args: [{
                    declarations: [ToastifyToastComponent, ToastifyToastContainerComponent],
                    imports: [common.CommonModule],
                    exports: [ToastifyToastContainerComponent]
                },] }
    ];

    /*
     * Public API Surface of angular-toastify
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.AngularToastifyModule = AngularToastifyModule;
    exports.ToastService = ToastService;
    exports.ToastifyToastContainerComponent = ToastifyToastContainerComponent;
    exports.ɵa = ToastifyToastComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
