// angular-toastify/esm2015/lib/toast.service.js
import { Injectable, EventEmitter } from '@angular/core';
import { Toast } from './toast';
import { ToastType } from './toast-type';
import * as i0 from "@angular/core";
export class ToastService {
    constructor() {
        this.toastAddedEvent = new EventEmitter();
        this.dismissAllEvent = new EventEmitter();
    }
    dismissAll() {
        this.dismissAllEvent.emit();
    }
    info(message) {
        const toast = new Toast(message, ToastType.info);
        this.toastAddedEvent.emit(toast);
    }
    success(message) {
        const toast = new Toast(message, ToastType.success);
        this.toastAddedEvent.emit(toast);
    }
    warn(message) {
        const toast = new Toast(message, ToastType.warning);
        this.toastAddedEvent.emit(toast);
    }
    error(message) {
        const toast = new Toast(message, ToastType.error);
        this.toastAddedEvent.emit(toast);
    }
    default(message) {
        const toast = new Toast(message, ToastType.default);
        this.toastAddedEvent.emit(toast);
    }
}
/** @nocollapse */ ToastService.ɵprov = i0.ɵɵdefineInjectable({ factory: function ToastService_Factory() { return new ToastService(); }, token: ToastService, providedIn: "root" });
ToastService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
ToastService.ctorParameters = () => [];
